File: pset1/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    float dollars;
    do
    {
        // asks users for height
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    // rounds it to nearest penny
    int cents = round(dollars * 100);
    // printf("cents: %i \n",cents);


    int counter = 0;

    // counts if cash is more than 25 than deducts 25 and ads 1 to the counter
    while (cents >= 25)
    {
        cents -= 25;
        counter += 1;
    }

    // counts if cash is more than 10 than deducts 10 and ads 1 to the counter
    while (cents >= 10)
    {
        cents -= 10;
        counter += 1;
    }

    // counts if cash is more than 5 than deducts 5 and ads 1 to the counter
    while (cents >= 5)
    {
        cents -= 5;
        counter += 1;
    }

    // counts if cash is more than 1 than deducts 1 and ads 1 to the counter
    while (cents >= 1)
    {
        cents -= 1;
        counter += 1;
    }
    // returns total counter as change owned
    printf("%i \n", counter);

}

File: pset6/python_Exercise/cash.py
from cs50 import get_float
import sys

while True:
    dollars = get_float("Change owed: ")
    if(dollars > 0):
        break

cents = round(dollars * 100)

counter = 0

# according to greedy algorithm counter if change is 25 or more
while cents >= 25:
    cents = cents - 25
    counter += 1
# according to greedy algorithm counter if change is 10 or more
while cents >= 10:
    cents = cents - 10
    counter += 1
# according to greedy algorithm counter if change is 5 or more
while cents >= 5:
    cents = cents - 5
    counter += 1
# according to greedy algorithm counter if change is 1 or more
while cents >= 1:
    cents = cents - 1
    counter += 1
# Prints total counter or lowest change owned
print(counter)